projects/js-packages/charts/src/components/line-chart/stories/annotation.docs.mdx

```diff
@@ -0,0 +1,472 @@
+			display: 'flex',
+			alignItems: 'center',
+			justifyContent: 'center',
+			borderRadius: '50%'
+		}}>
+			D
+		</span>
+	);
+
+	<LineChart.Annotation
+		datum={ dataPoint }
+		title="Deployed"
+		subjectType="line-vertical"
+		renderLabel={ () => (
+			<span style={ { transform: 'translate(0, 6px)' } }>
+				<DeployIcon />
+			</span>
+		) }
+	/>` }
+/>
+
+### Interactive Popovers
+
+Add rich interactive content with `renderLabelPopover`:
+
+<Source
+	language="jsx"
+	code={ `<LineChart.Annotation
+		datum={dataPoint}
+		title="Deployment"
+		renderLabel={() => <DeployIcon />}
+		renderLabelPopover={({ title, subtitle }) => (
+			<div style={{ display: 'flex', flexDirection: 'column', gap: '0.5rem' }}>
+				<div style={{
+					display: 'flex',
+					alignItems: 'center',
+					gap: '6px'
+				}}>
+					<DeployIcon />
+					<strong>Deploy finished</strong>
+				</div>
+				<p style={{ margin: 0 }}>
+					Thu. Apr 24, 2025. 09:57:23 UTC
+				</p>
+			</div>
+		)}
+	/>` }
+/>
+
+When both `renderLabel` and `renderLabelPopover` are provided, the label becomes clickable and opens the popover content.
+
+## Advanced Features
+
+### Smart Positioning
+
+**Basic annotations** (without custom `renderLabel`) automatically adjust their position to stay within chart boundaries:
+
+- **Horizontal overflow**: Labels flip to the left when approaching the right edge
+- **Vertical overflow**: Labels reposition above or below to avoid clipping
+- **Multi-line content**: Height is measured dynamically for accurate positioning
+
+**Note**: Smart positioning is **not available** for custom annotations using `renderLabel`. Custom annotations are positioned at the exact data point coordinates with `dx: 0, dy: 0` offset.
+
+### Label Positioning
+
+Control label placement with positioning options:
+
+<Source
+	language="jsx"
+	code={ `// Position at chart edges
+	<LineChart.Annotation
+		styles={{
+			label: {
+				x: 'start', // 'start', 'end', or number
+				y: 'start', // 'start', 'end', or number
+			}
+		}}
+	/>` }
+/>
+
+### Zero Value Support
+
+Annotations work correctly with zero values (unlike some chart libraries that treat `0` as falsy):
+
+<Source
+	language="jsx"
+	code={ `<LineChart.Annotation
+		datum={{ date: new Date('2024-01-01'), value: 0 }}
+		title="Baseline"
+		subtitle="Starting point"
+	/>` }
+/>
+
+## Accessibility
+
+### Keyboard Navigation
+
+- Annotations are focusable with Tab navigation
+- Interactive popovers can be opened with Enter or Space
+- Popovers can be closed with Escape key
+
+### Screen Reader Support
+
+- Interactive annotation buttons have `aria-label` attributes
+- Close buttons have descriptive labels
+- **Note**: Custom popover content does not currently have explicit ARIA roles or `aria-describedby` relationships. Content relies on the native Popover API's built-in accessibility features.
+
+### Focus Management
+
+- Focus returns to the annotation trigger when closed
+- Visual focus indicators are provided
```

Comment:
Will be updated after improvements in [CHARTS-59: Line Chart: Annotation accessibility](https://linear.app/a8c/issue/CHARTS-59/line-chart-annotation-accessibility)